exercises/move_semantics/move_semantics2.rs

```diff
@@ -5,15 +5,17 @@
 // Execute `rustlings hint move_semantics2` or use the `hint` watch subcommand
 // for a hint.
 
-// I AM NOT DONE
-
 #[test]
 fn main() {
     let vec0 = vec![22, 44, 66];
+    assert_eq!(vec0, vec![22, 44, 66]);
 
     let mut vec1 = fill_vec(vec0);
 
-    assert_eq!(vec0, vec![22, 44, 66]);
+    // the ownership of vec0 has passed to the argument of the function so it was removed so we can print vec0 here.
```

Comment:
yep, just send to the function a reference of vec0 and vec0 will stay with its value since the ownership is stay there and doesn't move to the function

Reply:
but I think I will need to do some "clone()" inside the function when I am defining there a new variable.

Reply:
```
fn main() {
    let vec0 = vec![22, 44, 66];
    // assert_eq!(vec0, vec![22, 44, 66]);

    let mut vec1 = fill_vec(&vec0);

    assert_eq!(vec0, vec![22, 44, 66]);
    assert_eq!(vec1, vec![22, 44, 66, 88]);
}

fn fill_vec(vec: &Vec<i32>) -> Vec<i32> {
    let mut vec = vec.clone();

    vec.push(88);

    vec
}
```